src/include/miopen/activ/problem_description.hpp

```diff
@@ -45,18 +45,35 @@ enum class Direction
 
 struct ProblemDescription
 {
-    ProblemDescription(Direction direction_,
-                       const ActivationDescriptor& activ,
+    // Forward constructor
+    ProblemDescription(const ActivationDescriptor& activ,
                        const TensorDescriptor& xDesc_,
                        const TensorDescriptor& yDesc_)
-        : direction(direction_), activDesc(activ), xDesc(xDesc_), yDesc(yDesc_)
+        : direction(Direction::Forward), activDesc(activ), xDesc(xDesc_), yDesc(yDesc_)
+    {
+    }
+
+    // Backward constructor
+    ProblemDescription(const ActivationDescriptor& activ,
+                       const TensorDescriptor& xDesc_,
+                       const TensorDescriptor& yDesc_,
+                       const TensorDescriptor& dxDesc_,
+                       const TensorDescriptor& dyDesc_)
+        : direction(Direction::Backward),
+          activDesc(activ),
+          xDesc(xDesc_),
+          yDesc(yDesc_),
+          dxDesc(dxDesc_),
+          dyDesc(dyDesc_)
     {
     }
 
     Direction GetDirection() const { return direction; }
     const ActivationDescriptor& GetActivDesc() const { return activDesc; }
     const TensorDescriptor& GetXDesc() const { return xDesc; }
     const TensorDescriptor& GetYDesc() const { return yDesc; }
+    const TensorDescriptor& GetDXDesc() const { return xDesc; }
```

Comment:
Should this check for the direction ?

Reply:
Can do that, but for bwd it's inited with {}.

Reply:
I think better way if we want to change it would be to make it boost::optional<TensorDescriptor>.

Reply:
@DrizztDoUrden Let's move forward. Please make _separate_ PR with these changes.